### DISCUSSION TITLE
True i18n for transactional emails

### DISCUSSION THREAD

**Discussion by nomocas** [clerk/javascript#195] [Support request - DO NOT OPEN HERE]
## Package

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend-core`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/edge`
- [ ] other:

## Description

Only one template (per email type - i.e. welcome, invitation, OTP, etc) is allowed for transactional emails.
We need to produce an app for several languages, and so we miss a way to manage them.

Any idea about some solutions and a roadmap?

Thanks.

**Comment by SokratisVidros** [comment]
Hi @nomocas. 

Yes. In 2022 we plan to give the ability to create custom email templates for your instances via [Clerk Email & SMS templates](https://clerk.dev/docs/how-to/email-sms-templates) feature.

Meanwhile, you can send transactional emails to your users in Clerk via the [Backend API](https://reference.clerk.dev/reference/backend-api-reference/emails).

**Comment by nomocas** [comment]
Hi @SokratisVidros,

Thanks for answering and great for the email API.

But... How could it be used to send an OTP code (with additional content) to a particular user **in his language** (by example)? 

When I check `@clerk/clerk-sdk-node` or `@clerk/backend-core`: I see the allowed payload type, which does not allow me to define which kind of email I want to send, nor the way to include an OTP code (there is nothing in the documentation you provide about a way to define a moustache-like template with variables substitution).

i.e.: The allowed payload:
```typescript
type EmailParams = {
  fromEmailName: string;
  emailAddressId: string;
  subject: string;
  body: string;
};
```

The response (aka the fields listed in the provided documentation):
```typescript
static attributes = ['id', 'fromEmailName', 'toEmailAddress', 'emailAddressId', 'subject', 'body', 'status'];
```

Also, if possible, it would mean that we need to rely on our backend to manage the authentication steps (e.g. send the OTP email itself - or magiclinks). And so it changes the way we should use react hooks in login related UI components (as it needs to call our backend in place of yours).

As I understand it, for the moment, the email API is only aimed to send other transactional emails than OTP code or magiclinks.

Am I wrong?

Any more precise idea about the roadmap for multiple templates via [Clerk Email & SMS templates](https://clerk.dev/docs/how-to/email-sms-templates) feature?

Thanks.

**Comment by braden-clerk** [comment]
Hey @nomocas, you're right, there's no way to have multiple versions of the "system emails" right now.  As Sokratis mentioned, this is slated for this year, but not this quarter.  We still need to look at priorities for Q3.

An MVP here would be the following:
- Adding a language attribute on a user, that defaults to "default", but is modifiable by you
- Adding a language modifier on each of the "system" emails, so that you can create versions for a different language.  
- When a system email gets sent, if the users language matches one of the variants you've created, send that one instead.

^ this sounds reasonable for us to build in a shorter time span.

The more complete feature set would include:
- Guess the best language for the User based on IP address
- Have a language selector on our UIs that lets the user change their language
- Already have translated system emails
- Already have translated UIs,
- etc.  We haven't done a full-planning for i18n quite yet.

**Comment by nomocas** [comment]
Hi @braden-clerk,

What you describe in your MVP is exactly what we need. 
We have a custom login component that is already translated, and language selection could not be based on IPs in our case, so all we need is to be able to assign a language to a user (which should of course define which template version to pick).

It would be awesome to have it quickly...

Where could we vote ? :)

Cheers.

**Comment by SokratisVidros** [comment]
@nomocas Your request has already been added to our backlog.

**Reply by KevyMbappeOld** [reply]
Any update on this?

**Reply by SokratisVidros** [reply]
Hey @FrankMbappe, This is currently scheduled for beginning of Q3 2022 that starts in July. How does your timeline look like?

**Reply by lavipta** [reply]
no update on this?

**Comment by jescalan** [comment]
https://feedback.clerk.com/roadmap/894efb80-f5e8-4258-8974-cfe3da12a4c8
